tests/unit/BloomFilterSyncTest.cpp

```diff
@@ -0,0 +1,127 @@
+/* This code is part of the GenSync project developed at Boston University.  Please see the README for use and references. */
+//
+// Created by Anish Sinha on 12/4/23.
+//
+
+#include "BloomFilterSyncTest.h"
+#include <GenSync/Syncs/GenSync.h>
+#include <GenSync/Syncs/BloomFilterSync.h>
+#include "../TestAuxiliary.h"
+CPPUNIT_TEST_SUITE_REGISTRATION(BloomFilterSyncTest);
+
+BloomFilterSyncTest::BloomFilterSyncTest() = default;
+
+BloomFilterSyncTest::~BloomFilterSyncTest() = default;
+
+void BloomFilterSyncTest::setUp() {
+	const int SEED = 93;
+	srand(SEED);
+}
+
+void BloomFilterSyncTest::tearDown() {
+}
+
+void BloomFilterSyncTest::BloomFilterSyncSetReconcileTest() {
+	const int BITS = sizeof(randZZ());
+	
+	GenSync GenSyncServer = GenSync::Builder().
+			setProtocol(GenSync::SyncProtocol::BloomFilterSync).
+			setComm(GenSync::SyncComm::socket).
+			setBits(BITS).
+			setExpNumElems(numExpElem).
+			setFalsePosProb(0.001).
+			build();
+	
+	GenSync GenSyncClient = GenSync::Builder().
+			setProtocol(GenSync::SyncProtocol::BloomFilterSync).
+			setComm(GenSync::SyncComm::socket).
+			setBits(BITS).
+			setExpNumElems(numExpElem).
+			setFalsePosProb(0.001).
+			build();
+	
+	//(oneWay = false, probSync = true, syncParamTest = false, Multiset = false, largeSync = false)
+	CPPUNIT_ASSERT(syncTest(GenSyncClient, GenSyncServer, false, true, false, false, false));
+}
+
+void BloomFilterSyncTest::BloomFilterSyncLargeSetReconcileTest(){
+	const int BITS = sizeof(randZZ());
+	
+	GenSync GenSyncServer = GenSync::Builder().
+			setProtocol(GenSync::SyncProtocol::BloomFilterSync).
+			setComm(GenSync::SyncComm::socket).
+			setBits(BITS).
+			setExpNumElems(largeNumExpElems).
+			setFalsePosProb(0.001).
+			build();
+	
+	GenSync GenSyncClient = GenSync::Builder().
+			setProtocol(GenSync::SyncProtocol::BloomFilterSync).
+			setComm(GenSync::SyncComm::socket).
+			setBits(BITS).
+			setExpNumElems(largeNumExpElems).
+			setFalsePosProb(0.001).
+			build();
+	
+	//(oneWay = false, probSync = true, syncParamTest = false, Multiset = false, largeSync = true)
+	CPPUNIT_ASSERT(syncTest(GenSyncClient, GenSyncServer, false, true,false,false,true));
+}
+
+void BloomFilterSyncTest::testAddElem(){
+	// number of elems to add
+	const int ITEMS = 50;
+	GenSync bfSync = GenSync::Builder().
+			setProtocol(GenSync::SyncProtocol::BloomFilterSync).
+			setComm(GenSync::SyncComm::socket).
+			setBits(4).
+			setExpNumElems(numExpElem).
+			setFalsePosProb(0.001).
+			build();
+	multiset<shared_ptr<DataObject>, cmp<shared_ptr<DataObject>>> elts;
+	
+	// add items works
+	for(int ii = 0; ii < ITEMS; ii++) {
+		shared_ptr<DataObject> item = make_shared<DataObject>(randZZ());
```

Comment:
The problem here is that you have no control over the order in which unit tests are run.  As such, you can have different random numbers being used in different tests.  I suggest, instead, that you seed the random number generator in the setup produced that is called before each test.

Reply:
Thanks for catching this, I added a SetSeed call to the setUp method ensuring that the same random ZZs will be generated each test run.